### ISSUE TITLE
Add Preview Link in PR comments

### ISSUE THREAD

**Issue by ericclemmons** [aws-amplify/amplify-console#264]
**Is your feature request related to a problem? Please describe.**

Without an explicit link being manually written into the PR description/body, there's friction for others to review what the PR does.

**Describe the solution you'd like**

When the PR preview is ready, I'd like for the app to comment on my PR with the preview link.  

**Additional context**

As an example, ZEIT does something similar:

> ![Screen Shot 2019-11-21 at 2 02 19 PM](https://user-images.githubusercontent.com/15182/69380563-8cb41b80-0c67-11ea-89e8-ec7676e3465c.png)
> – https://zeit.co/github#features

**Comment by swaminator** [comment]
@ericclemmons I know this is slightly different from what you've asked but there's already a way to view the URLs directly from the PR: https://docs.aws.amazon.com/amplify/latest/userguide/pr-previews.html#enable-web-previews

Click on **Details** for the Amplify Console Web Preview check, and you should see a link to your deployed URL. Is this good enough?

**Comment by ericclemmons** [comment]
@swaminator That's right, I used that guide to enable PR previews for a project.  

The dilemma is, how do collaborators on a repo (or outside contributors) know where to preview the preview?

Currently, it appears that someone with access to the Console has to log in, & copy/paste the URL back into GitHub.

**Comment by swaminator** [comment]
Incorrect. From the PR Click on Details for the Amplify Console Web Preview check, and you should see a link to your deployed URL.  Here's the image
![](https://docs.aws.amazon.com/amplify/latest/userguide/images/previews3.png)

**Comment by ericclemmons** [comment]
Yes, that works!  I should've been clearer on what I was looking for:

> When the PR preview is ready, I'd like for the app to **comment on my PR** with the preview link.

The problem we ran into was with visibility, not that functionality doesn't exist it. (https://zeit.co/github#features does a good job providing this visibility, though they do per-commit deploys)

Currently, we have an open PR:
> ![Screen Shot 2019-11-22 at 2 24 40 PM](https://user-images.githubusercontent.com/15182/69464721-fbf74180-0d33-11ea-822d-4734c67abc3a.png)

It's not visible at all to collaborators that there's a preview available, unless a collaborator clicks _Show all checks_ and then clicks the _Details_ link:

> ![Screen Shot 2019-11-22 at 2 24 45 PM](https://user-images.githubusercontent.com/15182/69464722-fbf74180-0d33-11ea-9a00-8ae89e65171c.png)

I'm thrilled that I was able to wire up PR previews in under an hour with Amplify Console!  I was hoping there were a way to make it clear when a PR was opened that reviewers could see the deployment front-and-center. 👀

Let me know if this makes sense, otherwise feel free to close. It's only a minor bit of friction & manual copy/pasting in each PR.

**Comment by HMilbradt** [comment]
Also noticed that this was missing.  Even our technical people didn't realize the links were there until I told them, and it's really easy to forget about considering how many other repos we have that don't have Amplify previews running.

**Comment by thedmeyer** [comment]
This feature would be a great enhancement to the PR Preview experience.

**Comment by patleeman** [comment]
+1 to this feature.

Now when clicking details you're taken to a /checks page and you need to click the "View more details on AWS Amplify (us-east-1) " link to see your preview URL. This is not really intuitive as I'd expect that to link to Amplify Console instead of my preview url.

![Screen Shot 2020-02-28 at 10 25 36 AM](https://user-images.githubusercontent.com/11035426/75561253-b2924e80-5a14-11ea-9da7-5fbd4ea23edf.png)

**Comment by Vivalldi** [comment]
I also agree that the preview link should be placed in a comment. Bitbucket users don't have an integrated app (that I'm aware of) to view the status. A comment upon preview deploy completion would suffice for our QA team to preview & approve the PR.

**Comment by fabidick22** [comment]
Any updates on this?

**Comment by richAtreides** [comment]
How do you make a github action trigger after completion of a successful deploy? I'm struggling to get this to trigger a ZAP OWASP scan after build finishes.

**Comment by daltonfury42** [comment]
There is a GitHub action that creates a preview branch and adds a comment to the PR with the link

https://github.com/yinlinchen/amplify-preview-actions

This is pretty useful as Amplify Preview feature is not supported for public repos.

**Comment by dwmkerr** [comment]
It would be great to have the option to have the Amplify Console comment on the pull request with the preview URL. At the moment, the flow is:

1. Find the status message and choose "Show all checks"
2. Choose "More details" in the AWS Amplify Console check
3. Find the link the AWS page showing the pull request (quite hard to see)
4. Open the link to AWS **potentially requiring auth** which can slow the flow down
5. Find the link to the preview URL and open it

This is reasonably manageable for a technical user, but for non-technical users it is quite hard to follow - and they might not even have access to the AWS Management Console.

I think adding the preview URL to the pull request is kind of essential to allow non-technical users to see that the preview is available. This preview feature is amazing by the way, it would just be great to do this last mile part of the work to make it more visible to users!

**Comment by PTaylour** [comment]
The preview link is predictable so as a _temporary_ workaround we use:


```yaml
name: Comment PR Preview

on:
  pull_request:
    branches:
      - main
jobs:
  comment-preview-link:
    runs-on: ubuntu-latest
    steps:
      - uses: mshick/add-pr-comment@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          message: |
            ## [🏗 Preview Build](https://pr-${{ github.event.number }}.<app_id>.amplifyapp.com/)
            ⏲ I just popped it in the oven, give it a few mins and it'll be ready!
          allow-repeats: false

```

**Comment by ericclemmons** [comment]
@PTaylour Thanks for sharing that GitHub Action!  I really like the messaging, too :D
> ⏲ I just popped it in the oven, give it a few mins and it'll be ready!

**Comment by PTaylour** [comment]
Looks like this has been implemented now! (eu-west-1)

<img width="395" alt="image" src="https://user-images.githubusercontent.com/3678476/135604641-9cd6b75a-7b0c-4d68-a3ee-efe35509b6cb.png">

**Comment by jogold** [comment]
> Looks like this has been implemented now! (eu-west-1)

Does someone know how to disable this comment? 😄

**Comment by swaminator** [comment]
This feature has been implemented. Sorry, there isn't a way to disable it now.

**Comment by jogold** [comment]
> This feature has been implemented. Sorry, there isn't a way to disable it now.

At least do something to delete previous comments before posting a new one...

**Comment by ericclemmons** [comment]
Do we need to re-install the GitHub app/bot to see this new behavior?

I did a quick test on a auto-branch deploy (#2239) and confirmed that #2239 is fixed, but didn't get a comment: https://github.com/aws-amplify/amplify-ui/pull/463

**Comment by charpeni** [comment]
@swaminator Our pull requests are getting spammed on each commit now, I do appreciate the visibility on Amplify Previews that this comment is bringing, but I would really appreciate a way to opt-out or at least if the message could be updated if one already existsst instead of adding a new one, that would be truly appreciated. 🙏

**Comment by dburles** [comment]
As other have mentioned, the comments are creating additional noise in PR's and we would also appreciate a way to opt-out  from them.

**Comment by naderhen** [comment]
In addition to consolidating these alerts, or removing outdated ones.. would it be possible to have the comments respect any domains/subdomains configured for the project? I'm seeing the nice subdomain links in the Amplify console, but the PR comments still display the old urls like 
![image](https://user-images.githubusercontent.com/99392/136077217-95264714-2b01-45e2-a7ea-b7b955a765ee.png)

**Comment by samkio** [comment]
For anyone else who found this issue when looking to disable the comments; here is the feature request to make this feature configurable https://github.com/aws-amplify/amplify-console/issues/2293
